testSuite/shared/src/test/scala/org/scalajs/testsuite/javalib/time/InstantTest.scala

```diff
@@ -424,4 +428,24 @@ class InstantTest extends TemporalTest[Instant] {
     expectThrows(classOf[DateTimeException], Instant.from(aYear))
   }
 
+  @Test def parse(): Unit = {
+    assertEquals(Instant.parse("1970-01-01T00:00:00Z"), Instant.EPOCH)
+    assertEquals(Instant.parse("-1000000000-01-01T00:00:00Z"), Instant.MIN)
+    assertEquals(Instant.parse("-999999999-01-01T00:00:00Z"), Instant.MIN.plus(366, DAYS))
+
+    // https://github.com/scala-js/scala-js-java-time/issues/23
+    assertEquals(Instant.parse("1970-01-01T00:10:00.100Z"), Instant.EPOCH.plus(10, MINUTES).plusMillis(100))
+
+    assertEquals(Instant.parse("+1000000000-12-31T23:59:59.999999999Z"), Instant.MAX)
+    assertEquals(Instant.parse("+999999999-12-31T23:59:59.999999999Z"), Instant.MAX.minus(366, DAYS))
+
+    assertEquals(Instant.parse("1999-06-03T06:56:23.942Z"), somePositiveInstant)
+    assertEquals(Instant.parse("-0687-08-07T23:38:33.088936253Z"), someNegativeInstant)
+
+
+    expectThrows(classOf[DateTimeParseException], Instant.parse("+1000000001-12-31T23:59:59.999999999Z"))
+    expectThrows(classOf[DateTimeParseException], Instant.parse("-0687-99-07T23:38:33.088936253Z"))
+    expectThrows(classOf[DateTimeParseException], Instant.parse("-ABCD-08-07T23:38:33.088936253Z"))
+  }
```

Comment:
The exceptional cases are under-tested here too. For example, what about `...T24:...` or `T13:65:...`?